### ISSUE TITLE
[explicit-module-boundary-types] add pattern option for args

### ISSUE THREAD

**Issue by bennypowers** [typescript-eslint/typescript-eslint#2686]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

`explicit-module-boundary-types` should provide an option to exclude method and function args by pattern, e.g.
```json
{
  "allowPattern": "/^_/"
}
```

**Repro**

```JSON
{
  "extends": [
      "plugin:@typescript-eslint/eslint-recommended",
      "plugin:@typescript-eslint/recommended",
  ]
}
```

```TS
import { FieldFunctionOptions, TypePolicy } from '@apollo/client/core';

export const Query: TypePolicy = {
  fields: {
    location(_, { args }: FieldFunctionOptions): Location {
      // function body is irrelevant
      return args as unknown as Location;
    },
  },
};
```

```json
{
  "include": ["src", "lib"],
  "exclude": ["node_modules", "**/node_modules/**/node_modules"],
  "compilerOptions": {
    "experimentalDecorators": true,
    "allowSyntheticDefaultImports": true,
    "resolveJsonModule": true,
    "moduleResolution": "Node",
    "target": "ES2019",
    "module": "ESNext",
    "noEmitOnError": false,
    "baseUrl": ".",
    "sourceMap": true,
    "plugins": [
      { "transform": "@zerollup/ts-transform-paths" },
      { "name": "ts-lit-plugin" }
    ],
  },
  "lib": [
    "ESNext.array",
    "ES2020",
    "ES2019.array",
    "ES2020.Intl.d.ts",
    "DOM",
    "DOM.iterable"
  ]
}
```

**Expected Result**

I expect unused params (`_`) should not trigger lint warnings

**Actual Result**

>  95:14  warning  Argument '_' should be typed  @typescript-eslint/explicit-module-boundary-types

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `^4.4.0` |
| `@typescript-eslint/parser`        | `^4.4.0` |
| `TypeScript`                       | `^4.0.3` |
| `ESLint`                           | `^7.11.0` |
| `node`                             | `12.17.0` |

**Comment by bradzacher** [comment]
This would be fixed by https://github.com/typescript-eslint/typescript-eslint/issues/2183

The problem here is you're having to unnecessarily annotate parameters.
